From d34fc7f1930e519da596e992f41a4b34f51f6a8f Mon Sep 17 00:00:00 2001
From: Wang ShaoBo <bobo.shaobowang@huawei.com>
Date: Wed, 6 Jul 2022 10:06:45 +0800
Subject: [PATCH 3297/4585] [Huawei] arm64/mpam: decrease dom_num when domain
 goes offline

Offering: HULK
hulk inclusion
category: bugfix
bugzilla: 187154

--------------------------------

mpam_resctrl_setup_domain() increase dom_num when domain goes online. As
a result, mpam_resctrl_cpu_offline() should decrease dom_num when domain
goes offline.

Signed-off-by: Wang ShaoBo <bobo.shaobowang@huawei.com>
Signed-off-by: Jialin Zhang <zhangjialin11@huawei.com>
---
 arch/arm64/kernel/mpam/mpam_setup.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/arch/arm64/kernel/mpam/mpam_setup.c b/arch/arm64/kernel/mpam/mpam_setup.c
index d30910e0cda2..6e71c99d19b0 100644
--- a/arch/arm64/kernel/mpam/mpam_setup.c
+++ b/arch/arm64/kernel/mpam/mpam_setup.c
@@ -172,6 +172,8 @@ int mpam_resctrl_cpu_offline(unsigned int cpu)
 		list_del(&d->list);
 		dom = container_of(d, struct mpam_resctrl_dom, resctrl_dom);
 		kfree(dom);
+
+		res->resctrl_res.dom_num--;
 	}
 
 	mpam_resctrl_clear_default_cpu(cpu);
-- 
2.17.1

